src/highdicom/legacy/sop.py

```diff
@@ -23,533 +24,385 @@
-            series_instance_uid=series_instance_uid,
-            series_number=series_number,
-            sop_instance_uid=sop_instance_uid,
-            sop_class_uid=sop_class_uid,
-            instance_number=instance_number,
-            manufacturer=ref_ds.Manufacturer,
-            modality=ref_ds.Modality,
-            transfer_syntax_uid=None,  # FIXME: frame encoding
-            patient_id=ref_ds.PatientID,
-            patient_name=ref_ds.PatientName,
-            patient_birth_date=ref_ds.PatientBirthDate,
-            patient_sex=ref_ds.PatientSex,
-            accession_number=ref_ds.AccessionNumber,
-            study_id=ref_ds.StudyID,
-            study_date=ref_ds.StudyDate,
-            study_time=ref_ds.StudyTime,
-            referring_physician_name=ref_ds.ReferringPhysicianName,
-            **kwargs
-        )
-        _convert_legacy_to_enhanced(legacy_datasets, self)
-
-
-class LegacyConvertedEnhancedPETImage(SOPClass):
-
+                distinguishing_tags_missing.append(tg)
+        logger_msg = set()
+        for ds in self.MixedFramesCopy:
+            all_equal = True
+            for tg in distinguishing_tags_missing:
+                if tg in ds:
+                    logger_msg.add(
+                        '{} is missing in all but {}'.format(
+                            DicomHelper.tag2kwstr(tg), ds['SOPInstanceUID']))
+                    all_equal = False
+                    break
+            if not all_equal:
+                continue
+            for tg in distinguishing_tags_existing:
+                ref_val = similar_ds[0][tg].value
+                if tg not in ds:
+                    all_equal = False
+                    break
+                new_val = ds[tg].value
+                if not DicomHelper.isequal(ref_val, new_val):
+                    logger_msg.add(
+                        'Inequality on distinguishing '
+                        'attribute{} -> {} != {} \n series uid = {}'.format(
+                            DicomHelper.tag2kwstr(tg), ref_val, new_val,
+                            ds.SeriesInstanceUID))
+                    all_equal = False
+                    break
+            if all_equal:
+                similar_ds.append(ds)
+        for msg_ in logger_msg:
+            logger.info(msg_)
+        for ds in similar_ds:
+            if ds in self.MixedFramesCopy:
+                self.MixedFramesCopy.remove(ds)
+        return (similar_ds, distinguishing_tags_existing)
+
+    @property
+    def DistinguishingAttributeKeywords(self) -> List[str]:
+        return self._DistinguishingAttributeKeywords[:]
+
+    @property
+    def FrameSets(self) -> List[FrameSet]:
+        return self._FrameSets
+
+
+class LegacyConvertedEnhanceImage(SOPClass):
     """SOP class for Legacy Converted Enhanced PET Image instances."""
 
     def __init__(
             self,
-            legacy_datasets: Sequence[Dataset],
+            frame_set: FrameSet,
```

Comment:
I understand your concern but based on use cases I ran into while converting tcia data collection, there are series with more than one multi-frame image. In order to have control over conversion, we need to separate them prior to conversion. This is the main reason that I used frameset collection externally. Calculating framesets internally will impose extra process overhead because in either case, I need to check the input dicoms externally to make sure I'm not feeding the SOPClass mixed frameset. Anyway, I'll modify the code as you prefer.

Reply:
@afshinmessiah do you have SOPSeriesInstanceUID for a sample series that has multiple multi-frame instances included?

Reply:
This series should have two multi-frames.
`SeriesInstanceUID = 1.3.6.1.4.1.14519.5.2.1.2783.4001.144515889821738216325254106222`
I must add that I tried to exclude them in the query, but it doesn't have anything to do with the code. Code should check how many instances there are and generate a separate UID for each.

Reply:
From what I see, all instances in that series have SOPClassUID of CTImageStorage, and I do not see any instance that has 2 frames (none has the "NumberOfFrames" attribute). Are you sure?

Reply:
From mixed series I didn't mean them having mixed SOPClassUID, I meant they have different distinguishing attributes which leads to different framsets.

Reply:
Thanks for the clarification. I misunderstood the initial comment.